src/apps/external_executor/protobuf/historical.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+import "kv.proto";
+
+package externalexecutor.protobuf;
+
+option optimize_for = LITE_RUNTIME;
+
+service Historical
+{
+  // Historical data look up
+  rpc GetHistoricalData(HistoricalData) returns (QueryResponse) {}
+}
+
+message HistoricalData
+{
+  string map_name = 1;
+  bytes key = 2;
+  int64 view = 3;
+  int64 seqno = 4;
+}
+
+message RetryQuery { bool retryquery = 1; }
+message QueryResponse
+{
+  externalexecutor.protobuf.KVValue data = 1;
+  RetryQuery retry = 2;
```

Comment:
Why is this a nested message, rather than an inline `bool`? It looks like the sample Python code only checks this top-level field (`result.HasField("retry")`), and the nested bool is unused.